service/storage_service.cc

```diff
@@ -2087,10 +2098,33 @@ future<> storage_service::decommission() {
                 throw;
             }
 
-            if (raft_available) {
-                slogger.info("DECOMMISSIONING: leaving Raft group 0");
-                ss._group0->leave_group0().get();
-                slogger.info("DECOMMISSIONING: left Raft group 0");
+            // Step 8: Leave group 0
+            //
+            // If the node failed to leave the token ring, don't remove it from group 0
+            // --- hence the `left_token_ring` check.
+            std::exception_ptr leave_group0_ex;
+            try {
+                if (raft_available && left_token_ring) {
```

Comment:
Isn't left_token_ring always true here? If we didn't leave the ring, we will exit via throwing.